### DISCUSSION TITLE
Get current filter state/applied filter values in embedded Superset dashboard

### DISCUSSION THREAD

**Discussion by vikashrajgupta** [apache/superset#28196] [Ideas]
Hi there,
After some research, I thought to share a feature request with you guys. I'm not able to get the current filter state or filter values applied by the public users on the embedded Superset Dashboard.

I checked the discussion and issues section but didn't find any solution for this, as someone also reported this [issues](https://github.com/apache/superset/discussions/25609) 9-10 months back.

As this API `api/v1/dashboard/45/filter_state/432zc83OZeli4v88QkNBwS2v0Q1xy_xByKePj_3p--RST52ZkkgKiHZjBT7xeTaB` provides the exact value as per my requirements but I want to fetch the same using an embedded dashboard that is available in Iframe.

That would be very helpful if you can look into this or let me know if its already exists and I have missed it.

**Comment by vikashrajgupta** [comment]
@rusackas - Do we have this functionality in Superset? I'm still not able to get a solution for this.
Can you please help me out?

**Reply by rusackas** [reply]
Not that I'm aware of. Filters are typically seen as a feature for dashboard consumers to use independently of one another, rather than for something that dashboard authors/embedders would use. That said, filter default values were kind of intended to scratch this itch, so you could configure those. 

When you change filters on a dashboard, you do see the `native_filters_key` change, which is effectively a "permalink" to that filter state. I'm not sure, however, if you can embed that permalink URL using the Embedded SDK though. Maybe @Vitor-Avila or @michael-s-molina can recall.

**Reply by Vitor-Avila** [reply]
@rusackas this isn't natively possible today.

**Reply by vikashrajgupta** [reply]
@Vitor-Avila - Are we planning to add such things? so that we can get the applied filter values through embedded SDK?

**Reply by vikashrajgupta** [reply]
@rusackas - You're right, and `native_filters_key` is useful when you're using the dashboard via domain but I require to fetch the filters inside Iframe.
Sometimes it's quite useful, like in my case, I have to perform some action when a user applies some filter.

**Reply by Vitor-Avila** [reply]
@vikashrajgupta I believe this might have been asked before, but I'm not aware of any Organization actively working to implement it. Contribution is always encouraged, tho :)

**Comment by d1manson** [comment]
Also keen on this - it would be really helpful to allow a user to share/bookmark a specific set of filters, even when embedded.

Maybe the permalink logic available in the non-embedded context could be leveraged via some message passing between the js sdk and the iframe, e.g. the sdk could offer a `getPermalink()` function that returns a promise, and a `startingPermalink` arguement to intialise it with.

**Comment by LiamTorrelli** [comment]
I burned a week on postMessage hooks and switchboard listeners trying to pull filter state out of an embedded dashboard (#34277 is the thread that sent me there). Sometimes it works. Usually you're racing the handshake — call too early and it throws, call late and you've already lost the user's selection.

Two problems get lumped together in threads like this:

**Host app sync** — your app can't read what's selected inside the iframe. Deep-linking "EMEA + last 30 days" from your own routes means Rison-encoded native filter IDs or fragile dataMask timing, not a clean callback. I wrote up all six embed limits I keep hitting: https://drafted.work/blog/superset-iframe-embedding-problems

**Saved combinations inside the dashboard** — if what you need is "land on this filter combo every visit without re-clicking," that's different. Upstream removed Filter Sets in 4.0. We rebuilt named presets with per-user defaults and drift sync when someone edits the dashboard: https://drafted.work/superset/dashboard-filter-presets

**Comment by rusackas** [comment]
Couple of relevant Preset blog posts on the topic... they've been out for a long time, but should still provide some useful reference. 
https://preset.io/blog/managing-filter-state-for-embedded-dashboards/
https://preset.io/blog/retrieving-the-current-filter-state-in-embedded-dashboards/
If those are out of date, or don't answer the questions herein, @Vitor-Avila and I can probably update them as needed.
